src/value/value.rs

```diff
@@ -31,7 +34,133 @@ mod serde;
 pub type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
 
 /// The storage mapping for the `Object` variant.
-pub type ObjectMap = BTreeMap<KeyString, Value>;
+// pub type ObjectMap = BTreeMap<KeyString, Value>;
+
+#[derive(Eq, PartialEq, Hash, Debug, Clone, Default, Serialize, Deserialize)]
+pub struct ObjectMap(pub Arc<BTreeMap<KeyString, Value>>);
```

Comment:
Would it make sense to call this just `Object` and keep the `ObjectMap` type alias for the underlying map type where it is needed? I know this is stooping to the level of bikeshedding…

However, if so, the newtype for the array below could simply be `struct Array`.